azurerm/modules/azurerm-adf/main.tf

```diff
@@ -0,0 +1,39 @@
+resource "azurerm_data_factory" "example" {
+  count                           = var.create_adf ? 1 : 0
+  name                            = var.resource_namer
+  location                        = var.resource_group_location
+  resource_group_name             = var.resource_group_name
+  public_network_enabled          = var.public_network_enabled
+  managed_virtual_network_enabled = var.managed_virtual_network_enabled
+
+  # optional block Adf identity 
+  dynamic "identity" {
+
+    for_each = var.adf_idenity ? toset([1]) : toset([0])
+    content {
+      type         = var.identity_type
+      identity_ids = var.identity_ids
+    }
+  }
+
+
+  #git configuration 
+  dynamic "github_configuration" {
```

Comment:
Looking at the [TF documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory), we might want here to give the option to set up ADF git integration with either `github_configuration` or `vsts_configuration` (depending if GH or Azure Repos is used at the client). Most of the options / vars will be shared across them.

Reply:
have made changes to enable either github or vsts